extjsdk/src/main/java/io/vantiq/extjsdk/ExtensionWebSocketClient.java

```diff
@@ -282,7 +282,9 @@ public void sendNotification(Object data) {
                 log.warn("Obtaining space to sent notifications was interrupted.", ie);
             } catch (Exception e) {
                 // If we get an exception during the send, we're unlikely to get a response so release now.
-                outstandingNotifications.release();
+                if (outstandingNotifications != null) {
```

Comment:
Is there a race condition here whereby `isConnected()` is true when we check, but false (& thus oustandingNotifications has gone away?

Is this a case where we should have a local version of `outstandingNotifications` that we get then then check that?  Do we need to check the `outstandingNotifications` reference above?

That is
```
if (isConnected()) {
    localoutstandingNotifications = outstandingNoficiations
....
    if (localOuut...) {
        localist....acquire()
   }
....
....  
       lockout.release()
```